walletconnectv2/src/main/kotlin/com/walletconnect/walletconnectv2/network/data/connection/controller/ConnectionController.kt

```diff
@@ -0,0 +1,25 @@
+package com.walletconnect.walletconnectv2.network.data.connection.controller
+
+import com.walletconnect.walletconnectv2.network.data.connection.ConnectionState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+internal sealed class ConnectionController {
+
+    internal class Manual : ConnectionController() {
+
+        private val _connectionState: MutableStateFlow<ConnectionState> = MutableStateFlow(ConnectionState.DISCONNECT)
+
+        val connectionStateFlow: StateFlow<ConnectionState> = _connectionState
+
+        fun connect() {
+            _connectionState.value = ConnectionState.CONNECT
+        }
+
+        fun disconnect() {
+            _connectionState.value = ConnectionState.DISCONNECT
```

Comment:
I believe this represents event more than state

```suggestion
        private val _connectionEvent: MutableStateFlow<ConnectionEvent> = MutableStateFlow(ConnectionEvent.DISCONNECT)

        val connectionEventFlow: StateFlow<ConnectionEvent> = _connectionEvent

        fun connect() {
            _connectionEvent.value = ConnectionEvent.CONNECT
        }

        fun disconnect() {
            _connectionEvent.value = ConnectionEvent.DISCONNECT
```